mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -449,6 +449,16 @@ class SparseVector(
   override def toString: String =
     "(%s,%s,%s)".format(size, indices.mkString("[", ",", "]"), values.mkString("[", ",", "]"))
 
```

Comment:
Yes, the nonzero entries idea did cross my mind. Maybe it's overcautious that I think it might become a complexity if we want to have another kind of Vector in the future, which don't have handy internal structure to scan for the nonzero entries. Again, this can be overcautious. 

And the dense == sparse idea looks good, maybe that suits into a util method better as it would not introduce the existence of `DenseVector` to `SparseVector` and vice versa.